### ISSUE TITLE
Train loss does not converge

### ISSUE THREAD

**Issue by WinterCyan** [WinterCyan/Gardner2019#1]
The implementation of network's final params output layers (d, l, s, c, a) maybe wrong.
Used Method: all 3 lights' **l**'s(3-dim direction) param is concat into one 9-dim tensor, and  other params are the same; which might caused some chaos during training.

**Comment by shapin94** [comment]
Hi, 

I'm currently working with the same paper, and facing with the same problem (Training loss does not converge)
Have you solve the problem? I've already double checked the paper and the implementation, but I couldn't find the reason.

**Comment by WinterCyan** [comment]
> Hi,
> 
> I'm currently working with the same paper, and facing with the same problem (Training loss does not converge)
> Have you solve the problem? I've already double checked the paper and the implementation, but I couldn't find the reason.

Sorry, I'm working on another project recently, the issue is still there (˘･_･˘). Perhaps you can ask the paper's authors for help😬

**Comment by cedar0817** [comment]
Hi,
I am working with this paper recently, I found the loss of training step1 with lighting coefficient 0.8 and ambient 0.2 in your training code, but the wr and wa is 20 and 1 respectively in original paper, can you explain it why？

**Comment by cedar0817** [comment]
> Hi,
> 
> I'm currently working with the same paper, and facing with the same problem (Training loss does not converge) Have you solve the problem? I've already double checked the paper and the implementation, but I couldn't find the reason.

Hi,
Have you solved this problem?
